src/main/java/seedu/address/ui/ArticleCard.java

```diff
@@ -69,5 +77,27 @@ public ArticleCard(Article article, int displayedIndex) {
                 .forEach(outlet -> outlets.getChildren().add(new Label(outlet.outletName)));
         publicationDate.setText(article.getPublicationDateAsString());
         status.setText(article.getStatus().toString());
+
+        String link = article.getLink().link;
+        if (!link.isEmpty()) {
+            hyperlink.setText("Link");
+            hyperlink.setOnAction(event -> {
+                openBrowser(link);
+            });
+        } else {
+            hyperlink.setVisible(false);
+        }
+    }
+
+    private void openBrowser(String link) {
+        if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
+            try {
+                Desktop.getDesktop().browse(new URI(link));
+            } catch (IOException | URISyntaxException e) {
+                e.printStackTrace();
```

Comment:
We should replace the `e.printStackTrace()` here with a more helpful exception to be thrown right? Since a user, upon entering an invalid URL will cause the program to terminate abruptly or not provide a meaningful error message to be shown to the user.